openhands/core/exceptions.py

```diff
@@ -77,3 +77,10 @@ def __init__(self, message='User cancelled the request'):
 class MicroAgentValidationError(Exception):
     def __init__(self, message='Micro agent validation failed'):
         super().__init__(message)
+
+
+class OperationCancelled(Exception):
+    """Exception raised when an operation is cancelled (e.g. by a keyboard interrupt)."""
+
+    def __init__(self, message='Operation was cancelled'):
```

Comment:
Just a question here, why aren't we still using UserCancelled, as previously?

Reply:
It's aimed to be future-proof: if `UserCancelled` were to be used e.g. for a UI button to cancel an operation, it is not the same in this context.

Reply:
If I see this right, UserCancelledError is not used, but in a test it is assumed to be what you describe.

Reply:
Ah, that's the async variant, which we don't use yet anyways. I'll change those occurences as well.

Reply:
I don't think we need to change them? Wait, so in my understanding, it was assumed in that test that it means "a UI button to cancel an operation", the kind of Stop Task coming from somewhere that doesn't shut down the app. Then it makes sense that we keep UserCancelledError _for that_. I thought that's what you said too.

Reply:
Yeh, I was just looking at the callback function, so those UserCancelledError occurences are actually desired, so no changes then. :)

Reply:
On a side note, I really think we'd be happier if we move the async and streaming code somehow, in a subclass or a mixin or something. I know they will be needed, but they currently aren't in use, and I don't know, it's just difficult to read llm.py 😢

Reply:
But not in this PR, right?